### PR TITLE
Task 7

### PR DESCRIPTION
**Pull request by alimovdavron**
All the tasks are implemented including one optional one. (Cognito task is not implemented)
1. authorization-service is added to the repo, has correct basicAuthorizer lambda and correct serverless.yaml file - [link to the serverless.yaml](https://github.com/alimovdavron/node-aws-be/blob/dff086f36bac9c27c02e08cc6e296344ba3945be/authorization-service/serverless.yml#L26-L28), [link to authorizer code](https://github.com/alimovdavron/node-aws-be/blob/dff086f36bac9c27c02e08cc6e296344ba3945be/authorization-service/src/functions/basicAuthorizer/handler.ts)
2. import-service serverless.yaml file has authorizer configuration for the importProductsFile lambda. Request to the importProductsFile lambda should work only with correct authorization_token being decoded and checked by basicAuthorizer lambda. Response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if Authorization header is not provided. - [link to the serverless.yaml](https://github.com/alimovdavron/node-aws-be/blob/dff086f36bac9c27c02e08cc6e296344ba3945be/import-service/serverless.yml#L53-L55)
3. update client application to send Authorization: Basic authorization_token header on import. Client should get authorization_token value from browser localStorage https://developer.mozilla.org/ru/docs/Web/API/Window/localStorage authorization_token = localStorage.getItem('authorization_token') - [link to the FE pull request](https://github.com/alimovdavron/shop-react-redux-cloudfront/pull/4)

Additional tasks: 
Client application should display alerts for the responses in 401 and 403 HTTP statuses. This behavior should be added to the nodejs-aws-fe-main/src/index.tsx file - please see [FE pull request](https://github.com/alimovdavron/shop-react-redux-cloudfront/pull/4) for the further information

### CODE REVIEW

File: authorization-service/package.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  },
+  "engines": {
+    "node": ">=14.15.0"
+  },
+  "dependencies": {
+    "@babel/preset-typescript": "^7.14.5",
+    "@middy/core": "^1.5.2",
+    "@middy/http-cors": "^2.4.1",
+    "@middy/http-json-body-parser": "^1.5.2",
+    "@types/jest": "^26.0.23",
```
Comment (noway36) [inline authorization-service/package.json:20]: Looks like we can move some `dependencies` to the `devDependencies` instead. And it seems there are some redundant dependencies such as `pg`

File: authorization-service/src/functions/basicAuthorizer/handler.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    authResponse.policyDocument = policyDocument;
+  }
+
+  return authResponse;
+}
+
+const deny = (methodArn) => generatePolicy('user', 'Deny', methodArn);
+const allow = (methodArn) => generatePolicy('user', 'Allow', methodArn);
+
+export const main = async (event, _, callback) => {
```
Comment (noway36) [inline authorization-service/src/functions/basicAuthorizer/handler.ts:25]: btw, there is a typing for this handler, called APIGatewayTokenAuthorizerHandler. It should be in `aws-lambda` package

**Comment by noway36** [comment]
LGTM 6/6

File: authorization-service/package.json (hunk 1 above)
Comment (alimovdavron) [reply authorization-service/package.json:20]: Yep, agree. I tried to handle these dependencies, to remove modules not related to shared module, but got webpack errors and give up without resolving :P

**Pull request merged**